packages/generator/src/action-function-import/action.ts

```diff
@@ -7,15 +7,29 @@ import { additionalDocForEntityNotDeserializable } from './function';
 
 const parameterName = 'parameters';
 
+function actionImportReturnType(actionImport: VdmActionImport) {
+  let type = actionImport.returnType.returnType;
+
+  if (actionImport.returnType.isCollection) {
+    type = `${type}[]`;
```

Comment:
I guess this is where `never[]` comes from?

Reply:
yes